pkg/codegen/go/gen.go

```diff
@@ -1119,6 +1150,78 @@ func (pkg *pkgContext) genPlainType(w io.Writer, name, comment, deprecationMessa
 	fmt.Fprintf(w, "}\n\n")
 }
 
+// Returns the default value used to perform "is default" checks.
+func (pkg *pkgContext) defaultComparisonValue(t schema.Type) string {
+	switch {
+	case schema.IsPrimitiveType(t):
+		return primitiveNilValue(t)
+	case isNilType(t):
+		return "nil"
+	default:
+		return strings.TrimSuffix(pkg.typeString(t), "Args") + "{}"
```

Comment:
The fact that we use `strings.TrimSuffix` here is a very unfortunate historical artifact, it certainly will be a stumbling block for reviewing this later on... Another argument in favor of reflect version.